### PR TITLE
Google login 

### PR DESCRIPTION
**Pull request by KaviHarjani**
The Google login inside the webview is broken as currently uses a proxy, this should provide a better user experience, as the user is not asked to input any details and connects through google with saved accounts

### CODE REVIEW

File: lib/webview.dart [hunk 1]
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
   late String paymentUrl = "'$_screenerHomeUrl/payment/capture/'";
+  late String googleUrl = "'$_screenerHomeUrl/login/flutter_auth/'";
```
Comment (manujagobind) [inline lib/webview.dart]: Change variable name to googleLoginUrl

File: lib/webview.dart [hunk 2]
```diff
@@ -126,7 +152,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
               },
               zoomEnabled: false,
               navigationDelegate: (NavigationRequest request) {
-                if (request.url.contains("premium")) {
+                if (request.url.endsWith('login/google/')) {
```
Comment (manujagobind) [inline lib/webview.dart]: Use double quotes with strings for consistency.

File: lib/webview.dart [hunk 3]
```diff
@@ -62,6 +65,29 @@ class _ScreenerAppState extends State<ScreenerApp> {
...
+        googleUser = "${user.displayName}";
+        var _googleLogin = {
+          'displayName': user.displayName,
+          'id': user.id,
+          'email': user.email
+        };
+        postFunction(googleUrl, jsonEncode(_googleLogin), requestMethod);
+      }
+    } catch (error) {
+      print(error);
```
Comment (manujagobind) [inline lib/webview.dart]: Let's not use print in production.

File: lib/webview.dart [hunk 4]
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
   late String paymentUrl = "'$_screenerHomeUrl/payment/capture/'";
```
Comment (manujagobind) [inline lib/webview.dart:22]: Are both double and single quotes needed in this string?

**Review by manujagobind — COMMENTED** [review]
Will this require a local screener back-end with the flutter branch to test?

File: lib/webview.dart (hunk 4 above)
Comment (KaviHarjani) [reply lib/webview.dart:22]: Yes

File: lib/webview.dart [hunk 5]
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
```
Comment (manujagobind) [inline lib/webview.dart]: https://www.screener.in must be in double quotes for consistency.

File: lib/webview.dart [hunk 6]
```diff
@@ -62,6 +65,26 @@ class _ScreenerAppState extends State<ScreenerApp> {
...
+  Future<void> _handleSignIn() async {
+    try {
+      var user = await _googleSignIn.signIn();
+      if (user != null) {
+        var authToken = await user.authentication;
+        var accessToken = authToken.accessToken;
+        postFunction(googleLoginUrl, jsonEncode({'access_token': accessToken}),
+            requestMethod);
+      }
+    } catch (error) {
```
Comment (manujagobind) [inline lib/webview.dart:83]: Handle the exception.

Comment (KaviHarjani) [reply lib/webview.dart:83]: What should we do in this case? show an alert box? or just pass the post request and let the function give the error?

Comment (manujagobind) [reply lib/webview.dart:83]: Not too sure what to do here. 

We can display a "failed to login" or a generic message maybe but not sure if it's required.

**Review by manujagobind — APPROVED** [review]

**Comment by manujagobind** [comment]
Let's address any pending comments + add fix for 403 error and get this merged.

**Comment by KaviHarjani** [comment]
> Let's add

Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work
It would require changing google auth API, from my key to the screeners key

**Comment by manujagobind** [comment]
> > Let's add
> 
> Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work It would require changing google auth API, from my key to the screeners key

Where do we maintain the auth key?

**Comment by KaviHarjani** [comment]
> > > Let's add
> > 
> > 
> > Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work It would require changing google auth API, from my key to the screeners key
> 
> Where do we maintain the auth key?

we've to make a key locally, and have the SHA copied into the google console

**Pull request merged**